packages/squiggle-lang/__tests__/ReducerProject/ReducerProject_test.res

```diff
@@ -0,0 +1,174 @@
+})
+
+test("test bindings", () => {
+  let project = Project.createProject()
+  Project.setSource(project, "variables", "myVariable=666")
+  runFetchBindings(project, "variables")->expect->toBe("@{myVariable: 666}")
+})
+
+describe("project1", () => {
+  let project = Project.createProject()
+  Project.setSource(project, "first", "x=1")
+  Project.setSource(project, "main", "x")
+  Project.setContinues(project, "main", ["first"])
+
+  test("runOrder", () => {
+    expect(Project.getRunOrder(project)) == ["first", "main"]
+  })
+  test("dependents first", () => {
+    expect(Project.getDependents(project, "first")) == ["main"]
+  })
+  test("dependencies first", () => {
+    expect(Project.getDependencies(project, "first")) == []
+  })
+  test("dependents main", () => {
+    expect(Project.getDependents(project, "main")) == []
+  })
+  test("dependencies main", () => {
+    expect(Project.getDependencies(project, "main")) == ["first"]
+  })
+  test("test result", () => {
+    runFetchResult(project, "main")->expect->toBe("Ok(1)")
+  })
+  test("test bindings", () => {
+    runFetchBindings(project, "main")->expect->toBe("@{x: 1}")
+  })
+})
+
+describe("project2", () => {
+  let project = Project.createProject()
+  Project.setContinues(project, "main", ["second"])
+  Project.setContinues(project, "second", ["first"])
+  Project.setSource(project, "first", "x=1")
+  Project.setSource(project, "second", "y=2")
+  Project.setSource(project, "main", "y")
+
+  test("runOrder", () => {
+    expect(Project.getRunOrder(project)) == ["first", "second", "main"]
+  })
+  test("runOrderFor", () => {
+    expect(Project.getRunOrderFor(project, "first")) == ["first"]
+  })
+  test("dependencies first", () => {
+    expect(Project.getDependencies(project, "first")) == []
+  })
+  test("dependents first", () => {
+    expect(Project.getDependents(project, "first")) == ["second", "main"]
+  })
+  test("dependents main", () => {
+    expect(Project.getDependents(project, "main")) == []
+  })
+  test("dependencies main", () => {
+    expect(Project.getDependencies(project, "main")) == ["first", "second"]
+  })
+  test("test result", () => {
+    runFetchResult(project, "main")->expect->toBe("Ok(2)")
+  })
+  test("test bindings", () => {
+    runFetchBindings(project, "main")->expect->toBe("@{x: 1,y: 2}")
+  })
+})
+
+describe("project with include", () => {
+  let project = Project.createProject()
+  Project.setContinues(project, "main", ["second"])
+  Project.setContinues(project, "second", ["first"])
+
+  Project.setSource(
+    project,
+    "first",
+    `
+  #include 'common'
+  x=1`,
+  )
+  Project.parseIncludes(project, "first")
+  Project.parseIncludes(project, "first") //The only way of setting includes
+  //Don't forget to parse includes after changing the source
+
+  Project.setSource(project, "common", "common=0")
+  Project.setSource(
+    project,
+    "second",
+    `
+  #include 'common'
```

Comment:
Have we decided yet if this should refer to the full filename, the relative filename, or something else?